wfl/Models/StatusSearchCriteria.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using MongoDB.Driver;
+using System.Threading.Tasks;
+
+namespace Symphono.Wfl.Models
+{
+    public class StatusSearchCriteria: ICriteria<FoodOrder>
+    {
+        public FoodOrder.StatusOptions Status { get; set;}
+
+        public async Task<IEnumerable<FoodOrder>> ApplyCriteria(IMongoCollection<FoodOrder> collection)
+        {
+            IAsyncCursor<FoodOrder> task = await collection.FindAsync(o => o.Status == this.Status);
+            IList<FoodOrder> orders = task.ToList();
```

Comment:
I would personally expose the IAsyncCursor<T> from this interface, then we don't need to do any copy/paste of code.